### PR TITLE
Fix Support for new Email address domain #4547

### PR DESCRIPTION
**Pull request by elachino**
### Description
Fix support for all new domain in email address validation. This solution can be apply to rewrite all Zend_Validate::is with ZendFramework2. I propose only change to fix email validation in this PR.
You have to upgrade zend-validator module to last version from 2.5 to up, or you can get from magento 2.2.

### Fixed Issues (if relevant)

1. magento/magento2#4547: Support for new .tech TLD and others required

### Manual testing scenarios
1. try to recover a password from an email with some new domain like .shop or .srl

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=11576) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=11576) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: sidolov<br/>:x: elachino

**Comment by orlangur** [comment]
> You have to upgrade zend-validator module to last version from 2.5 to up, or you can get from magento 2.2.

This should be done in scope of this PR: https://github.com/magento/magento2/blob/2.1-develop/composer.json#L17 Commit both `composer.json` and `composer.lock`. Please use least version suitable, `2.5.x` should be fine probably.

File: app/code/Magento/Config/Model/Config/Backend/Email/Address.php [hunk 1]
```diff
@@ -20,7 +20,8 @@ class Address extends \Magento\Framework\App\Config\Value
     public function beforeSave()
     {
         $value = $this->getValue();
-        if (!\Zend_Validate::is($value, 'EmailAddress')) {
+		$validator = new \Zend\Validator\EmailAddress();
+        if(!$validator->isValid($value)){
```
Comment (orlangur) [inline app/code/Magento/Config/Model/Config/Backend/Email/Address.php]: Please fix formatting in all changed lines. 4 spaces must be used as indentation, there should be a space after `if` etc.

Comment (orlangur) [reply app/code/Magento/Config/Model/Config/Backend/Email/Address.php]: Great 👍 Besides formatting changes look good to me.

Please also fix `\Magento\Newsletter\Controller\Subscriber\NewAction::validateEmailFormat`.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by elachino** [comment]
I update with your requests, please re-check

**Comment by orlangur** [comment]
@elachino, https://github.com/magento/magento2/pull/11576/files - formatting is still not fixed, `composer.lock` is not committed.

Please check integration and static tests failures on Travis also, they must be green for this PR to be processed.

After all changes are made and all builds are green, please squash all changes into single commit and force push to the same branch.

File: composer.lock [hunk 2]
```diff
@@ -2003,7 +2003,7 @@
                 "zendframework/zend-servicemanager": "~2.4.0",
                 "zendframework/zend-session": "~2.4.0",
                 "zendframework/zend-text": "~2.4.0",
-                "zendframework/zend-validator": "~2.4.0",
+                "zendframework/zend-validator": "^2.6",
```
Comment (orlangur) [inline composer.lock]: Did you change this file manually? I don't see `content-hash` updated.

**Comment by elachino** [comment]
Yes, i do.
I copy only different row from my repository.
If you prefer i update with composer.


Alessandro Scavella | Co-founder @COODER | E-commerce & Mobile | Cooder.it <http://cooder.it/> | Personal number: 327/5431980




____________________________________________________
Avviso di riservatezza
questo messaggio è destinato unicamente alla persona o al soggetto ai quali è indirizzato e può contenere informazioni riservate e/o coperte da segreto professionale, la cui divulgazione è proibita. Qualora non siate i destinatari designati non dovrete leggere, utilizzare, diffondere o copiare le informazioni trasmesse (D.Lgs n.196/2003). Nel caso aveste ricevuto questo messaggio per errore, vogliate cortesemente contattare il mittente e cancellare il materiale dai vostri computer.
Notice of confidentiality
this message is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged information, the disclosure of which is prohibited. If you are not the intended recipient you may not read, use, disseminate or copy the information transmitted. If you have received this message in error, please contact the sender and delete the material from any computer.

**Comment by orlangur** [comment]
@elachino, yeah, it does not work this way, just run `composer update zendframework/zend-validator` with your updated `composer.json`.

**Comment by ihor-sviziev** [comment]
Hi @orlangur  do you need any help in reviewing this PR?

**Review by orlangur — APPROVED** [review]
I wasn't sure `composer require zendframework/zend-validator:^2.6.0` can actually work on `2.1-develop`.

Now I checked it worked, conflict in `composer.lock` can be fixed on our side.

**Review by orlangur — APPROVED** [review]

**Comment by josefbehr** [comment]
@elachino please resolve the merge conflict

**Comment by orlangur** [comment]
@josefbehr not needed, it's better to be resolved on our side as I mentioned.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by elachino** [comment]
I'm out for holiday, when return on 21 august I try to resolve it.
Kind regards

Il lun 13 ago 2018, 17:35 Josef Behr <notifications@github.com> ha scritto:

**Comment by ihor-sviziev** [comment]
Hi @elachino,
I tried to merge conflicts in your PR, but got an issue that another zendframework/zend-* packages has dependencies to zendframework/zend-validator: "~2.5", so actually we can't update it without updating another modules.

Could you take a look on it and merge 2.1-develop branch?

**Comment by magento-engcom-team** [comment]
Hi @elachino. Thank you for your contribution. 
We will aim to release these changes as part of 2.1.16. 
Please check the release notes for final confirmation.

**Pull request merged**